connector/src/main/scala/com/microsoft/kusto/spark/utils/CslCommandsGenerator.scala

```diff
@@ -81,22 +82,23 @@ private[kusto] object CslCommandsGenerator {
     s""".show table ${tableName} details | project todynamic(ShardingPolicy).UseShardEngine"""
   }
 
-  def generateTableMoveExtentsCommand(sourceTableName: String, destinationTableName: String, batchSize: Int,
+  def generateTableMoveExtentsCommand(sourceTableName: String, destinationTableName: String, timerange: Array[Instant], batchSize: Int,
                                       isDestinationTableMaterializedViewSource: Boolean = false): String = {
     val setNewIngestionTime: String = if (isDestinationTableMaterializedViewSource) "with(SetNewIngestionTime=true)" else ""
     s""".move extents to table $destinationTableName $setNewIngestionTime <|
-       .show table $sourceTableName extents with(extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":$batchSize}');
+       .show table $sourceTableName extents with(extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":$batchSize}') with(extentCreatedOnFrom='${timerange(0)}', extentCreatedOnTo='${timerange(1)}');
```

Comment:
Hi Yoni,

Yes we have E2E tests that checks the functionality by writing the data and reading that back again to check if both are equal. And the above code passed them. 

After your suggestion above i.e. `.move extents to table $destinationTableName with(setNewIngestionTime=<true of false>, extentCreatedOnFrom='...', extentCreatedOnTo='...', extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":...}') <| .show table $sourceTableName extents`, 
got the below findings:

1) I took the final command built by code (in current scenario) and ran it manually which ran without any issues with the desired result. 
```kql
.move async extents to table KustoSparkReadWriteTest_f84ac15e_3f51_49cf_a6f0_bbb70915xxxx  <|
       .show table sparkTempTable_KustoSink_KustoSparkReadWriteTest_xxxxx_3f51_49cf_a6f0_bbb70915018f_3004f1b6_1599_49a8_91cc_xxxxx
       extents with(extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":1}')
       with(extentCreatedOnFrom='2023-03-02T11:28:41.474Z', extentCreatedOnTo='2023-03-02T11:29:02.405063200Z');
```

2) After making the changes asked above, i.e. moving all the props in a single with we got exceptions:
```kql
.move async extents to table KustoSparkReadWriteTest_f84ac15e_3f51_49cf_a6f0_bbb70915xxxx with ( extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":1}', extentCreatedOnFrom='2023-03-02T11:28:41.474Z', extentCreatedOnTo='2023-03-02T11:29:02.405063200Z') <|
       .show table sparkTempTable_KustoSink_KustoSparkReadWriteTest_xxxxx_3f51_49cf_a6f0_bbb70915018f_3004f1b6_1599_49a8_91cc_xxxxx
       extents
```
 got the below error:
![image](https://user-images.githubusercontent.com/102726227/223050659-999d9259-0775-40a1-8d77-6abda4a73df5.png)


Which make sense to me, since we cannot move prop `extentsShowFilteringRuntimePolicy` before `.show table extents` command. 

Really appreciate your suggestions, or to wrap this faster we can have a 15min call?

Reply:
sorry for the confusion.

instead of this:

```
.move async extents to table KustoSparkReadWriteTest_f84ac15e_3f51_49cf_a6f0_bbb70915xxxx with ( extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":1}', extentCreatedOnFrom='2023-03-02T11:28:41.474Z', extentCreatedOnTo='2023-03-02T11:29:02.405063200Z') <|
       .show table sparkTempTable_KustoSink_KustoSparkReadWriteTest_xxxxx_3f51_49cf_a6f0_bbb70915018f_3004f1b6_1599_49a8_91cc_xxxxx
       extents
```

it should be this:

```
.move async extents to table KustoSparkReadWriteTest_f84ac15e_3f51_49cf_a6f0_bbb70915xxxx with (extentCreatedOnFrom='2023-03-02T11:28:41.474Z', extentCreatedOnTo='2023-03-02T11:29:02.405063200Z') <|
       .show table sparkTempTable_KustoSink_KustoSparkReadWriteTest_xxxxx_3f51_49cf_a6f0_bbb70915018f_3004f1b6_1599_49a8_91cc_xxxxx
       extents with(extentsShowFilteringRuntimePolicy='{"MaximumResultsCount":1}')
```

Reply:
Thanks @y0nil. Have added the changes.